apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryBigNumeric.cs

```diff
@@ -51,40 +42,38 @@ public struct BigQueryBigNumeric : IEquatable<BigQueryBigNumeric>, IComparable<B
         /// <summary>
         /// The maximum valid value for <see cref="BigQueryBigNumeric"/>, equal to (2^255 - 1)e-38.
         /// </summary>
-        public static BigQueryBigNumeric MaxValue { get; } = new BigQueryBigNumeric(s_maxValue);
+        public static BigQueryBigNumeric MaxValue { get; } = new BigQueryBigNumeric(s_numericType.MaxValue);
 
         /// <summary>
         /// The minimum valid value for <see cref="BigQueryBigNumeric"/>, equal to -2^255e-38.
         /// </summary>
-        public static BigQueryBigNumeric MinValue { get; } = new BigQueryBigNumeric(s_minValue);
+        public static BigQueryBigNumeric MinValue { get; } = new BigQueryBigNumeric(s_numericType.MinValue);
 
         /// <summary>
         /// The smallest positive value for <see cref="BigQueryBigNumeric"/>, equal to 1e-38.
         /// </summary>
-        public static BigQueryBigNumeric Epsilon { get; } = new BigQueryBigNumeric(1);
+        public static BigQueryBigNumeric Epsilon { get; } = new BigQueryBigNumeric(s_numericType.Epsilon);
 
-        // Integer representation, always scaled by 38 decimal places - so a value of 1 here represents
-        // a numeric value of 1e-38 for example.
-        private readonly BigInteger _integer;
+        // General purpose numeric representation. Should always be of type s_numericType unless
+        // it is the default value.
+        private readonly NumericValue _numericValue;
 
-        private BigQueryBigNumeric(BigInteger integer)
+        private BigQueryBigNumeric(NumericValue value)
         {
-            if (!IsRawValueInRange(integer))
-            {
-                throw new OverflowException("BigNumeric value out of range");
-            }
-            _integer = integer;
+            // This should never throw, it means there's an error on our code.
+            // At some point we maint want to accept NumericValues with a different NumericType
+            // as long as they are in range. But there's no need for that right now.
+            GaxPreconditions.CheckState(value == default || value.Type.Equals(s_numericType), $"Incorrect NumericType used to build a {nameof(BigQueryBigNumeric)}.");
```

Comment:
Here and elsewhere: don't use an interpolated string literal as an argument, or we're constructing an extra string every time we create a BigQueryBigNumeric right now. Admittedly I *think* in C# 10 it ends up counting as a constant string anyway, but I'm not absolutely sure. I think it would be simplest to just hardcode the string to be honest - alternatively, pass a format string as one argument and the nameof value as another.